packages/integration/src/bitbucket/core.ts

```diff
@@ -35,8 +35,16 @@ export async function getBitbucketDefaultBranch(
   const branchUrl = isHosted
     ? `${config.apiBaseUrl}/repositories/${project}/${repoName}`
     : `${config.apiBaseUrl}/projects/${project}/repos/${repoName}/default-branch`;
-
-  const response = await fetch(branchUrl, getBitbucketRequestOptions(config));
+    
+  var response = await fetch(branchUrl, getBitbucketRequestOptions(config));
+  
+  if (response.status === 404) {
```

Comment:
```suggestion
  if (response.status === 404 && !isHosted) {
```

In case of `bitbucket.org` this block shouldn't be executed.

Reply:
@OrkoHunter . added above check and tested it successfully ( own hosted) .